misk-hibernate/src/main/kotlin/misk/hibernate/ReflectionQueryFactory.kt

```diff
@@ -840,18 +916,80 @@ internal class ReflectionQuery<T : DbEntity<T>>(
               continue
             }
 
+            if (property.aggregation != AggregationType.NONE) {
+              errors.addAll(
+                collectAggregationTypeErrors(
+                  aggregation = property.aggregation,
+                  elementType = parameter.type.typeLiteral(),
+                  propertyName = "${projectionClass.java.name}#${parameter.name}",
+                ))
+              if (errors.isNotEmpty()) {
+                continue
+              }
+            }
+
             val path = (pathPrefix + property.path).split('.')
-            properties.add(path)
+            properties.add(ParsedProperty(path, property.aggregation))
           }
+          if (errors.isNotEmpty()) return
           selectMethodHandler = SelectMethodHandler(isList, constructor, properties)
         } else {
-          val onlyPath = select.path.split('.')
+          val onlyPath = ParsedProperty(select.path.split('.'), select.aggregation)
           selectMethodHandler = SelectMethodHandler(isList, null, listOf(onlyPath))
         }
 
         val javaMethod = function.javaMethod ?: throw UnsupportedOperationException()
         result[javaMethod] = selectMethodHandler
       }
+
+      private fun collectAggregationTypeErrors(
+        aggregation: AggregationType,
+        elementType: TypeLiteral<*>,
+        propertyName: String,
+      ) : MutableList<String> {
+        // Primitives are special. We have to ensure we are using the boxed types.
+        val doubleType = Double::class.createType(nullable = true).typeLiteral().rawType
```

Comment:
move these to a companion?

Reply:
[c82a0c1](https://github.com/cashapp/misk/pull/3237/commits/c82a0c151ef7d1f95a3837d21f153c50f92ce6dd)